### PR TITLE
Updated for Spree 3.2.0.rc2 and Rails 5

### PR DESCRIPTION
**Pull request by pikender**
Removed Gemfile.lock as should not be part of gem
Fixed deprecation warning uniq -> distinct
    See: https://github.com/rails/rails/pull/20198
Fixed exception for validate_presence “The record was indeed invalid, but it produced these validation errors instead”
    See: https://github.com/thoughtbot/shoulda-matchers/issues/861#issuecomment-189563425
Fixed deprecation warning - use keyword syntax for get, put etc in controller tests
Fixed coffee-rails not loading error

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-gem 'spree', github: 'spree/spree'
-gem 'spree_auth_devise', github: 'spree/spree_auth_devise'
-gem 'sass-rails', '~> 5.0'
+gem 'spree', github: 'spree/spree', branch: '3-2-stable'
+gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: 'master'
 
-group :assets do
-  gem 'coffee-rails', '4.0.1'
-end
+gem 'byebug'
```
Comment (tanmay3011) [inline Gemfile]: @pikender Please can we lock the versions here

File: spree_favorite_products.gemspec [hunk 2]
```diff
@@ -17,7 +17,9 @@ Gem::Specification.new do |s|
   s.require_path = 'lib'
   s.requirements << 'none'
 
-  s.add_dependency 'spree_core', '~> 3.2.0.beta1'
+  s.add_dependency 'spree_core', '~> 3.2.0.rc2'
+  s.add_development_dependency 'coffee-rails'
```
Comment (tanmay3011) [inline spree_favorite_products.gemspec]: :ditto:

File: Gemfile [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
 
 group :test do
   gem 'rspec-rails', '~> 3.5.2'
   gem 'shoulda-matchers', '3.1.1'
   gem 'simplecov', '~> 0.12.0', require: false
   gem 'database_cleaner', '~> 1.5.3'
   gem 'rspec-activemodel-mocks', '~> 1.0.3'
+  gem 'rails-controller-testing'
```
Comment (tanmay3011) [inline Gemfile]: :ditto:

**Comment by pikender** [comment]
@tanmay3011 changes done. Please have a look.

**Pull request merged**